subcommand/mesh-init/command.go

```diff
@@ -235,9 +240,32 @@ func (c *Command) realRun() error {
 			if err != nil {
 				c.log.Error("Error marking the status of checks as critical: %s", err.Error())
 			}
-		case <-c.ctx.Done():
-			// TODO: Handle consul logout and service/proxy deregistration once dataplane shuts down
-			return nil
+		case <-c.dataplaneMonitor.done():
+			var result error
+			c.log.Info("Dataplane has successfully shutdown. Deregistering services and terminating control plane")
+
+			err = c.deregisterServiceAndProxy(consulClient, clusterARN, serviceRegistration, proxyRegistration)
+			if err != nil {
+				c.log.Error("error deregistering service and proxy %s", err.Error())
+				result = multierror.Append(result, err)
+			}
+
+			if c.config.ConsulLogin.Enabled {
+				// TODO: Add unit tests for ACL Logout
+				_, err = consulClient.ACL().Logout(nil)
+				if err != nil {
+					c.log.Error("error logging out of consul %s", err.Error())
+					result = multierror.Append(result, err)
+				}
+			}
+
+			// Cleanup bootstrap directory
+			err = os.RemoveAll(c.config.BootstrapDir)
+			if err != nil {
+				c.log.Error("error cleaning up bootstrap directory %s", err.Error())
+				result = multierror.Append(result, err)
+			}
```

Comment:
Why remove the bootstrap directory? I feel like we can skip this step. The task is stopping and the data in the volume will be removed by ECS anyway, so we shouldn't need to do it ([ref1](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/fargate-task-storage.html), [ref2](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/bind-mounts.html)). What do you think?